compiler/src/iree/compiler/Pipelines/Options.h

```diff
@@ -36,23 +36,17 @@ struct InputDialectOptions {
     none,
     // Analyses the input to determine what input dialect pipeline to use.
     auto_detect,
-#ifdef IREE_HAVE_MHLO_INPUT
```

Comment:
Is it really needed?

Reply:
Yeah, those defines should remain. If you build the compiler without an input type, the option for that type should be not available.